pkg/operator/workloadcontroller/workload_controller.go

```diff
@@ -1,295 +0,0 @@
-		targetOperandVersion:  targetOperandVersion,
-		operatorImagePullSpec: operatorImagePullSpec,
-
-		operatorClient:        operatorClient,
-		versionRecorder:       versionRecorder,
-		operatorConfigClient:  operatorConfigClient,
-		openshiftConfigClient: openshiftConfigClient,
-		kubeClient:            kubeClient,
-		nodeInformer:          nodeInformer,
-		eventRecorder:         eventRecorder.WithComponentSuffix("workload-controller"),
-
-		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "OpenShiftAPIServerOperator"),
-	}
-
-	c.countNodes = c.countNodesImpl
-
-	operatorConfigInformer.Informer().AddEventHandler(c.eventHandler())
-	kubeInformersForEtcdNamespace.Core().V1().ConfigMaps().Informer().AddEventHandler(c.eventHandler())
-	kubeInformersForEtcdNamespace.Core().V1().Secrets().Informer().AddEventHandler(c.eventHandler())
-	kubeInformersForOpenShiftAPIServerNamespace.Core().V1().ConfigMaps().Informer().AddEventHandler(c.eventHandler())
-	kubeInformersForOpenShiftAPIServerNamespace.Core().V1().ServiceAccounts().Informer().AddEventHandler(c.eventHandler())
-	kubeInformersForOpenShiftAPIServerNamespace.Core().V1().Services().Informer().AddEventHandler(c.eventHandler())
-	kubeInformersForOpenShiftAPIServerNamespace.Apps().V1().Deployments().Informer().AddEventHandler(c.eventHandler())
-	kubeInformersForKubeSystemNamespace.Core().V1().ConfigMaps().Informer().AddEventHandler(c.eventHandler())
-	kubeInformersForOpenShiftConfigNamespace.Core().V1().ConfigMaps().Informer().AddEventHandler(c.eventHandler())
-	configInformers.Config().V1().Images().Informer().AddEventHandler(c.eventHandler())
-	apiregistrationInformers.Apiregistration().V1().APIServices().Informer().AddEventHandler(c.eventHandler())
-
-	// we only watch some namespaces
-	kubeInformersForOpenShiftAPIServerNamespace.Core().V1().Namespaces().Informer().AddEventHandler(c.namespaceEventHandler())
-
-	return c
-}
-
-func (c OpenShiftAPIServerOperator) sync() error {
-	operatorConfig, err := c.operatorConfigClient.OpenShiftAPIServers().Get("cluster", metav1.GetOptions{})
-	if err != nil {
-		return err
-	}
-
-	switch operatorConfig.Spec.ManagementState {
-	case operatorsv1.Managed:
-	case operatorsv1.Unmanaged:
-		return nil
-	case operatorsv1.Removed:
-		// TODO probably need to watch until the NS is really gone
-		if err := c.kubeClient.CoreV1().Namespaces().Delete(operatorclient.TargetNamespace, nil); err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-		return nil
-	default:
-		c.eventRecorder.Warningf("ManagementStateUnknown", "Unrecognized operator management state %q", operatorConfig.Spec.ManagementState)
-		return nil
-	}
-
-	kubeAPIServerOperator, err := c.openshiftConfigClient.ClusterOperators().Get("kube-apiserver", metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		kubeAPIServerOperator, err = c.openshiftConfigClient.ClusterOperators().Get("openshift-kube-apiserver-operator", metav1.GetOptions{})
```

Comment:
the generic workload controller is missing that line, it only checks for the existence of `kube-apiserver`